ddtrace/commands/ddtrace_run.py

```diff
@@ -105,13 +98,6 @@ def main():
 
     log.debug("program executable: %s", executable)
 
-    try:
-        # Raises OSError for permissions errors in Python 2
-        #        PermissionError for Python 3
-        os.execl(executable, executable, *args.command[1:])
-    except (OSError, PermissionError):
-        print("ddtrace-run: executable '%s' does not have executable permissions.\n" % executable)
-        parser.print_usage()
-        sys.exit(1)
+    os.execl(executable, executable, *args.command[1:])
```

Comment:
I'm wondering whether we could do something in between like:
~~~ python
    ...
except (OSError, PermissionError) as e:
    print("ddtrace-run: error on launch: %s" % e)
    ...
~~~

Reply:
Then we can also do
```python
    ...
except Exception as e:
    print("ddtrace-run: error on launch: %s" % e)
    ...
```

I'm not sure this is adding a lot of value, but at least it does not try to be smarter than the user 😆

Reply:
if we catch the exception, we can print to add context, but probably best to re-raise it

Reply:
I would prefer handling the known cases gracefully and only throw a traceback to the user's face when something totally unexpected happens, but that's just my preference 🙂 .

Reply:
Not sure what's "known cases". Paraphrasing every `errno`/`strerror` does not add any value.

Reply:
"known case" here is either `OSError` or `PermissionError`, not just a specific `errno`.

Reply:
Interesting! So I actually think the issue in this case isn't the fact that we didn't print the error code!

The usage of `ddtrace-run` is that you prefix it to your normal Python command (eg. `python file.py`). So in @jd's case it was actually expected behaviour because if he had done `./file.py` it would have failed as well with `zsh: permission denied: ./file.py`, or similar. (I think `pyddprofile` worked differently by allowing you to exclude `python` which is probably why @jd ran into this issue).

I had added this handling here because the most common misusage of `ddtrace-run` was exactly what @jd did but the usage message wasn't clear enough to correct the usage - which is what I think the problem actually is in this case!

This is what `ddtrace-run` did before the way it is now:

```
$ ddtrace-run file.py
Traceback (most recent call last):
  File "/private/tmp/.venv3.7/bin/ddtrace-run", line 8, in <module>
    sys.exit(main())
  File "/private/tmp/.venv3.7/lib/python3.7/site-packages/ddtrace/commands/ddtrace_run.py", line 82, in main
    os.execl(executable, executable, *sys.argv[2:])
  File "/Users/kyle.verhoog/.pyenv/versions/3.7.7/lib/python3.7/os.py", line 542, in execl
    execv(file, args)
PermissionError: [Errno 13] Permission denied
```

This is what is does now:

```
$ ddtrace-run file.py
ddtrace-run: executable 'file.py' does not have executable permissions: [Errno 13] Permission denied

usage: ddtrace-run <your usual python command>
```

The problem with it is probably the `<your usual python command>`. I think if we had a few examples like `python file.py`, `gunicorn file.py` and `file.py (if executable)` underneath then it'd be much more helpful.


Now with all that being said, I think @jd brings up a good point that we shouldn't hide the error information from the user (especially those that don't trust us 😆). The intention here was certainly not to outsmart the user but rather to be more helpful than just throwing an error at them. I think the solution is to do both: print the error as well as the usage information.